### PR TITLE
[W6.4d][T09-B3]LIWEN LAI

### PR DESCRIPTION
**Pull request by Livian1107**
add a method in command class, to judge whether the command can mutate the data

### CODE REVIEW

File: src/seedu/addressbook/commands/AddCommand.java [hunk 1]
```diff
@@ -12,7 +12,7 @@
  * Adds a person to the address book.
  */
 public class AddCommand extends Command {
-
+    public static final boolean type = true;
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/commands/AddCommand.java:15]: Think why you define this variable as above, e.g. I don't think it should be defined as `public`. Also for this particular case, I don't think you need a new constant (you write it as variable though).

File: src/seedu/addressbook/commands/AddCommand.java [hunk 2]
```diff
@@ -67,4 +67,9 @@ public CommandResult execute() {
         }
     }
 
+    @Override
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/commands/AddCommand.java:70]: Good job for using this annotation.

File: src/seedu/addressbook/commands/Command.java [hunk 3]
```diff
@@ -43,6 +43,13 @@ public CommandResult execute(){
         throw new UnsupportedOperationException("This method should be implement in child classes");
     }
 
+    /**
+     * judge whether the command will mutate the data
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/commands/Command.java:47]: This method `returns`, not `judges`. Need to rephrase the header comment, but good job for writing it.

File: src/seedu/addressbook/commands/Command.java [hunk 4]
```diff
@@ -43,6 +43,13 @@ public CommandResult execute(){
...
+     */
+    public boolean isMutating(){
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/commands/Command.java:49]: This method should defined as `abstract` and then you can remove the exception.

File: src/seedu/addressbook/commands/ListCommand.java [hunk 5]
```diff
@@ -22,4 +22,9 @@ public CommandResult execute() {
         List<ReadOnlyPerson> allPersons = addressBook.getAllPersons().immutableListView();
         return new CommandResult(getMessageForPersonListShownSummary(allPersons), allPersons);
     }
+
+    @Override
+    public boolean isMutating(){
+        return type;
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/commands/ListCommand.java:28]: No need to use a constant here.

**Comment by jeffryhartanto** [comment]
@Livian1107 Some comments added. Please close the PR after reading comments.

IMPORTANT: You need to update the Logic and LogicTest.

File: src/seedu/addressbook/commands/AddCommand.java (hunk 1 above)
Comment (Livian1107) [reply src/seedu/addressbook/commands/AddCommand.java:15]: I thought it is similar to magic number. We should avoid use 'true' directly so I set a new variable

**Pull request closed without merging**